### PR TITLE
docs: add note about "/" in name

### PR DESCRIPTION
**Pull request by prokopsimek**
I have spent a few hours debugging this. Please, add the note to the readme.

### CODE REVIEW

File: docs/usage/configuration.md [hunk 1]
```diff
@@ -76,6 +76,8 @@ The branches on which releases should happen. By default **semantic-release** wi
 
 **Note**: Once **semantic-release** is configured, any user with the permission to push commits on one of those branches will be able to publish a release. It is recommended to protect those branches, for example with [GitHub protected branches](https://help.github.com/articles/about-protected-branches).
 
+**Note**: The prerelease can't be named with `/`. For example `{ "name": "debug/*", "prerelease": true }` won't work. You have to manually remove `/` (example: `{ "name": "debug/*", "prerelease": '${name.replace(/\//, "-")}' }`)
+
```
Comment (gr2m) [inline docs/usage/configuration.md]: I wonder if this is rather a bug that could be fixed in code instead of adding a note about it?

Comment (prokopsimek) [reply docs/usage/configuration.md]: @gr2m I am wondering you're probably true. [Git allows slash in tags](https://stackoverflow.com/a/26382358/10826693). I'll try it fix it in another branch. I'd close this proposal.

**Comment by pvdlg** [comment]
What is the problem you are trying to solve? Do you have an error message?
CAn explain what you are trying to achieve?

**Comment by prokopsimek** [comment]
@pvdlg I apologize for the poor description. I didn't realize it could be a bug.

My `.releaserc`:
```json
{
  "plugins": [
    "@semantic-release/commit-analyzer",
    "@semantic-release/release-notes-generator",
    "@semantic-release/changelog",
    "@semantic-release/npm",
    "@semantic-release/git",
    "@semantic-release/github",
    [
      "semantic-release-slack-bot",
      {
        "notifyOnSuccess": true,
        "notifyOnFail": true,
        "markdownReleaseNotes": true
      }
    ]
  ],
  "branches": [
    "master",
    "next",
    { "name": "beta", "prerelease": "beta" },
    { "name": "alpha", "prerelease": "alpha" },
    { "name": "feature/*", "prerelease": true },
    { "name": "hotfix/*", "prerelease": true },
    { "name": "debug/*", "prerelease": true }
  ]
}
```

If I run `semantic-release -d` so it throws: 
```
EPRERELEASEBRANCH A pre-release branch configuration is invalid in the `branches` configuration.
```

<details>
<summary>Expand to see detailed log</summary>

semantic-release -d
yarn run v1.21.1
$ semantic-release -d
[18:01:29] [semantic-release] › ℹ  Running semantic-release version 16.0.1
[18:01:31] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/changelog"
[18:01:31] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/npm"
[18:01:31] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/git"
[18:01:31] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/github"
[18:01:31] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "semantic-release-slack-bot"
[18:01:31] [semantic-release] › ✔  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[18:01:31] [semantic-release] › ✔  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
[18:01:31] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/changelog"
[18:01:31] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/npm"
[18:01:31] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/git"
[18:01:31] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/npm"
[18:01:31] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/github"
[18:01:31] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/npm"
[18:01:31] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/github"
[18:01:31] [semantic-release] › ✔  Loaded plugin "success" from "@semantic-release/github"
[18:01:31] [semantic-release] › ✔  Loaded plugin "success" from "semantic-release-slack-bot"
[18:01:31] [semantic-release] › ✔  Loaded plugin "fail" from "@semantic-release/github"
[18:01:31] [semantic-release] › ✔  Loaded plugin "fail" from "semantic-release-slack-bot"
[18:02:11] [semantic-release] › ⚠  Skip step "fail" of plugin "@semantic-release/github" in dry-run mode
[18:02:11] [semantic-release] › ⚠  Skip step "fail" of plugin "semantic-release-slack-bot" in dry-run mode
[18:02:11] [semantic-release] › ✖  EPRERELEASEBRANCH A pre-release branch configuration is invalid in the `branches` configuration.
Each pre-release branch in the branches configuration (https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#branches) must have a prerelease property valid per the Semantic Versioning Specification (https://semver.org/#spec-item-9). If the prerelease property is set to true, then the name property is used instead.

Your configuration for the problematic branch is { name: 'debug/semver-release-config2', prerelease: true, tags: [ { gitTag: 'v1.0.0', version: '1.0.0', channels: [Array] }, { gitTag: 'v1.1.0', version: '1.1.0', channels: [Array] }, { gitTag: 'v1.1.1', version: '1.1.1', channels: [Array] }, { gitTag: 'v1.1.2', version: '1.1.2', channels: [Array] }, { gitTag: 'v1.1.3', version: '1.1.3', channels: [Array] }, ... 77 more items ] }.

[18:02:11] [semantic-release] › ✖  EPRERELEASEBRANCH A pre-release branch configuration is invalid in the `branches` configuration.
Each pre-release branch in the branches configuration (https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#branches) must have a prerelease property valid per the Semantic Versioning Specification (https://semver.org/#spec-item-9). If the prerelease property is set to true, then the name property is used instead.

Your configuration for the problematic branch is { name: 'debug/semantic-beta-releases', prerelease: true, tags: [ { gitTag: 'v1.0.0', version: '1.0.0', channels: [Array] }, { gitTag: 'v1.1.0', version: '1.1.0', channels: [Array] }, { gitTag: 'v1.1.1', version: '1.1.1', channels: [Array] }, { gitTag: 'v1.1.2', version: '1.1.2', channels: [Array] }, { gitTag: 'v1.1.3', version: '1.1.3', channels: [Array] }, ... 76 more items ] }.

[18:02:11] [semantic-release] › ✖  EPRERELEASEBRANCH A pre-release branch configuration is invalid in the `branches` configuration.
Each pre-release branch in the branches configuration (https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#branches) must have a prerelease property valid per the Semantic Versioning Specification (https://semver.org/#spec-item-9). If the prerelease property is set to true, then the name property is used instead.

Your configuration for the problematic branch is { name: 'hotfix/failing-thinPullRequestsPractice', prerelease: true, tags: [ { gitTag: 'v1.0.0', version: '1.0.0', channels: [Array] }, { gitTag: 'v1.1.0', version: '1.1.0', channels: [Array] }, { gitTag: 'v1.1.1', version: '1.1.1', channels: [Array] }, { gitTag: 'v1.1.2', version: '1.1.2', channels: [Array] }, { gitTag: 'v1.1.3', version: '1.1.3', channels: [Array] }, ... 76 more items ] }.

[18:02:11] [semantic-release] › ✖  EPRERELEASEBRANCH A pre-release branch configuration is invalid in the `branches` configuration.
Each pre-release branch in the branches configuration (https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#branches) must have a prerelease property valid per the Semantic Versioning Specification (https://semver.org/#spec-item-9). If the prerelease property is set to true, then the name property is used instead.

Your configuration for the problematic branch is { name: 'hotfix/add-filtering-pagination-services', prerelease: true, tags: [ { gitTag: 'v1.0.0', version: '1.0.0', channels: [Array] }, { gitTag: 'v1.1.0', version: '1.1.0', channels: [Array] }, { gitTag: 'v1.1.1', version: '1.1.1', channels: [Array] }, { gitTag: 'v1.1.2', version: '1.1.2', channels: [Array] }, { gitTag: 'v1.1.3', version: '1.1.3', channels: [Array] }, ... 76 more items ] }.

[18:02:11] [semantic-release] › ✖  EPRERELEASEBRANCH A pre-release branch configuration is invalid in the `branches` configuration.
Each pre-release branch in the branches configuration (https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#branches) must have a prerelease property valid per the Semantic Versioning Specification (https://semver.org/#spec-item-9). If the prerelease property is set to true, then the name property is used instead.

Your configuration for the problematic branch is { name: 'feature/pr-template', prerelease: true, tags: [ { gitTag: 'v1.0.0', version: '1.0.0', channels: [Array] }, { gitTag: 'v1.1.0', version: '1.1.0', channels: [Array] }, { gitTag: 'v1.1.1', version: '1.1.1', channels: [Array] }, { gitTag: 'v1.1.2', version: '1.1.2', channels: [Array] }, { gitTag: 'v1.1.3', version: '1.1.3', channels: [Array] }, ... 76 more items ] }.

[18:02:11] [semantic-release] › ✖  EPRERELEASEBRANCH A pre-release branch configuration is invalid in the `branches` configuration.
Each pre-release branch in the branches configuration (https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#branches) must have a prerelease property valid per the Semantic Versioning Specification (https://semver.org/#spec-item-9). If the prerelease property is set to true, then the name property is used instead.

Your configuration for the problematic branch is { name: 'feature/pr-comment', prerelease: true, tags: [ { gitTag: 'v1.0.0', version: '1.0.0', channels: [Array] }, { gitTag: 'v1.1.0', version: '1.1.0', channels: [Array] }, { gitTag: 'v1.1.1', version: '1.1.1', channels: [Array] }, { gitTag: 'v1.1.2', version: '1.1.2', channels: [Array] }, { gitTag: 'v1.1.3', version: '1.1.3', channels: [Array] }, ... 60 more items ] }.

[18:02:11] [semantic-release] › ✖  EPRERELEASEBRANCH A pre-release branch configuration is invalid in the `branches` configuration.
Each pre-release branch in the branches configuration (https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#branches) must have a prerelease property valid per the Semantic Versioning Specification (https://semver.org/#spec-item-9). If the prerelease property is set to true, then the name property is used instead.

Your configuration for the problematic branch is { name: 'feature/list-practices', prerelease: true, tags: [ { gitTag: 'v1.0.0', version: '1.0.0', channels: [Array] }, { gitTag: 'v1.1.0', version: '1.1.0', channels: [Array] }, { gitTag: 'v1.1.1', version: '1.1.1', channels: [Array] }, { gitTag: 'v1.1.2', version: '1.1.2', channels: [Array] }, { gitTag: 'v1.1.3', version: '1.1.3', channels: [Array] }, ... 76 more items ] }.

[18:02:11] [semantic-release] › ✖  EPRERELEASEBRANCH A pre-release branch configuration is invalid in the `branches` configuration.
Each pre-release branch in the branches configuration (https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#branches) must have a prerelease property valid per the Semantic Versioning Specification (https://semver.org/#spec-item-9). If the prerelease property is set to true, then the name property is used instead.

Your configuration for the problematic branch is { name: 'feature/fat-pullrequests-practice', prerelease: true, tags: [ { gitTag: 'v1.0.0', version: '1.0.0', channels: [Array] }, { gitTag: 'v1.1.0', version: '1.1.0', channels: [Array] }, { gitTag: 'v1.1.1', version: '1.1.1', channels: [Array] }, { gitTag: 'v1.1.2', version: '1.1.2', channels: [Array] }, { gitTag: 'v1.1.3', version: '1.1.3', channels: [Array] }, ... 60 more items ] }.

[18:02:11] [semantic-release] › ✖  EPRERELEASEBRANCH A pre-release branch configuration is invalid in the `branches` configuration.
Each pre-release branch in the branches configuration (https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#branches) must have a prerelease property valid per the Semantic Versioning Specification (https://semver.org/#spec-item-9). If the prerelease property is set to true, then the name property is used instead.

Your configuration for the problematic branch is { name: 'feature/ci-reporter', prerelease: true, tags: [ { gitTag: 'v1.0.0', version: '1.0.0', channels: [Array] }, { gitTag: 'v1.1.0', version: '1.1.0', channels: [Array] }, { gitTag: 'v1.1.1', version: '1.1.1', channels: [Array] }, { gitTag: 'v1.1.2', version: '1.1.2', channels: [Array] }, { gitTag: 'v1.1.3', version: '1.1.3', channels: [Array] }, ... 63 more items ] }.

AggregateError: 
    SemanticReleaseError: A pre-release branch configuration is invalid in the `branches` configuration.
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/get-error.js:6:10)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:37:21
        at Array.forEach (<anonymous>)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:35:26
        at Array.reduce (<anonymous>)
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:34:46)
        at runMicrotasks (<anonymous>)
        at async run (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:56:22)
        at async module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:257:22)
    SemanticReleaseError: A pre-release branch configuration is invalid in the `branches` configuration.
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/get-error.js:6:10)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:37:21
        at Array.forEach (<anonymous>)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:35:26
        at Array.reduce (<anonymous>)
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:34:46)
        at runMicrotasks (<anonymous>)
        at async run (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:56:22)
        at async module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:257:22)
    SemanticReleaseError: A pre-release branch configuration is invalid in the `branches` configuration.
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/get-error.js:6:10)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:37:21
        at Array.forEach (<anonymous>)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:35:26
        at Array.reduce (<anonymous>)
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:34:46)
        at runMicrotasks (<anonymous>)
        at async run (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:56:22)
        at async module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:257:22)
    SemanticReleaseError: A pre-release branch configuration is invalid in the `branches` configuration.
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/get-error.js:6:10)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:37:21
        at Array.forEach (<anonymous>)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:35:26
        at Array.reduce (<anonymous>)
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:34:46)
        at runMicrotasks (<anonymous>)
        at async run (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:56:22)
        at async module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:257:22)
    SemanticReleaseError: A pre-release branch configuration is invalid in the `branches` configuration.
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/get-error.js:6:10)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:37:21
        at Array.forEach (<anonymous>)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:35:26
        at Array.reduce (<anonymous>)
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:34:46)
        at runMicrotasks (<anonymous>)
        at async run (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:56:22)
        at async module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:257:22)
    SemanticReleaseError: A pre-release branch configuration is invalid in the `branches` configuration.
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/get-error.js:6:10)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:37:21
        at Array.forEach (<anonymous>)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:35:26
        at Array.reduce (<anonymous>)
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:34:46)
        at runMicrotasks (<anonymous>)
        at async run (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:56:22)
        at async module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:257:22)
    SemanticReleaseError: A pre-release branch configuration is invalid in the `branches` configuration.
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/get-error.js:6:10)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:37:21
        at Array.forEach (<anonymous>)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:35:26
        at Array.reduce (<anonymous>)
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:34:46)
        at runMicrotasks (<anonymous>)
        at async run (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:56:22)
        at async module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:257:22)
    SemanticReleaseError: A pre-release branch configuration is invalid in the `branches` configuration.
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/get-error.js:6:10)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:37:21
        at Array.forEach (<anonymous>)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:35:26
        at Array.reduce (<anonymous>)
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:34:46)
        at runMicrotasks (<anonymous>)
        at async run (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:56:22)
        at async module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:257:22)
    SemanticReleaseError: A pre-release branch configuration is invalid in the `branches` configuration.
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/get-error.js:6:10)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:37:21
        at Array.forEach (<anonymous>)
        at /Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:35:26
        at Array.reduce (<anonymous>)
        at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:34:46)
        at runMicrotasks (<anonymous>)
        at async run (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:56:22)
        at async module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:257:22)
    at module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/lib/branches/index.js:66:11)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async run (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:56:22)
    at async module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/index.js:257:22)
    at async module.exports (/Users/prokop/Sites/_dxheroes/scanner/node_modules/semantic-release/cli.js:55:5) {
  name: 'AggregateError'
error Command failed with exit code 1.
</details>

If I change the config to:

```
{
  "plugins": [
    "@semantic-release/commit-analyzer",
    "@semantic-release/release-notes-generator",
    "@semantic-release/changelog",
    "@semantic-release/npm",
    "@semantic-release/git",
    "@semantic-release/github",
    [
      "semantic-release-slack-bot",
      {
        "notifyOnSuccess": true,
        "notifyOnFail": true,
        "markdownReleaseNotes": true
      }
    ]
  ],
  "branches": [
    "master",
    "next",
    { "name": "beta", "prerelease": "beta" },
    { "name": "alpha", "prerelease": "alpha" },
    { "name": "feature/*", "prerelease": '${name.replace(/\//, "pre-")}' },
    { "name": "hotfix/*", "prerelease": '${name.replace(/\//, "pre-")}' },
    { "name": "debug/*", "prerelease": '${name.replace(/\//, "pre-")}' }
  ]
}
```
...so it works.

*I expect the new tag with `vX.Y.Z-debug/my-any-branch-name.1` is created.*

**Comment by pvdlg** [comment]
That's actually expected. `debug/my-any-branch-name` is not a valid pre-release id per semantic-version spec. That would create versions like `1.0.0-debug/my-any-branch-name.1` which is not valid.

I don't think we should says anything else in the doc other than when you use `prerelease: true` to generate the prerelease id based on the branch it has to be valid per the spec. We should't mention a workaround for `/` or anything else specific that would make the prerelease id invalid because there is no way to figure out all the possible to make it invalid.

**Comment by prokopsimek** [comment]
@pvdlg We develop a CLI tool and I'd like to ensure that each feature version works on Windows either Unix based systems so I'd like to create a tag named according to my branch. It means a format `vX.Y.Z-{{EXACT BRANCH NAME}}.1`. 

I understand it is an invalid semantic release format but is the semantic-release a tool that should dictate the rules? 
Wouldn't it be better to allow it? 
Could I create a PR for that?

**Comment by pvdlg** [comment]
As mentioned in the error it's invalid per the spec we follow ([Semantic Versioning Specification](https://semver.org/#spec-item-9)):
> Each pre-release branch in the branches configuration (https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#branches) must have a prerelease property valid per the Semantic Versioning Specification (https://semver.org/#spec-item-9). If the prerelease property is set to true, then the name property is used instead.

We don't want to diverge from the spec we follow. And you won't be able to publish such version on any package manager that follow the spec anyway.

**Comment by pvdlg** [comment]
> On the other hand, the error could be invalid if I have a { name: "feature*", prerelease: true } and my branch is named according to Git flow (feature/some-branch-name) because the branches configuration is correct then.

I don't understand. Are you saying there is a case where the error message will be wrong? Do you have an example?

**Comment by prokopsimek** [comment]
@pvdlg I'm sorry, the `feature*` is invalid glob according to Semver. The previous comment isn't true.

File: docs/usage/configuration.md (hunk 1 above)
Comment (prokopsimek) [reply docs/usage/configuration.md]: It's not a bug it's a feature. 🤣

**Comment by pvdlg** [comment]
If you use `feature*` then it will matches branches on your repo and use the name of those branches as the prerelease id. So it depends on the name of your branches.
Basically if you use the name of the branch to generate the prerelease id, then the name of the branch has to be a valid prerelease id. I believe this is what the error message says.

**Comment by prokopsimek** [comment]
> If you use `feature*` then it will matches branches on your repo and use the name of those branches as the prerelease id. So it depends on the name of your branches.
> Basically if you use the name of the branch to generate the prerelease id, then the name of the branch has to be a valid prerelease id. I believe this is what the error message says.

It's a detail but...

example 1:
my config: `{ name: "feature*", prerelease: true }`
my branch name: `feature-branch-name`
prerelease works?: ✅ 

example 2:
my config: `{ name: "feature*", prerelease: true }`
my branch name: `feature/branch-name` (used slash)
prerelease works?: 🚫
error: `EPRERELEASEBRANCH A pre-release branch configuration is invalid in the `branches` configuration.`

Is my configuration really invalid? No... I guess my branch naming convention is different.

**Comment by pvdlg** [comment]
> Is my configuration really invalid? No... I guess my branch naming convention is different.

But you use the branch name to generate the config...That's what the glob do.
As far as I can tell the error happens when it should and it's message is descriptive.
I don't really understand what the problem is here?

**Comment by prokopsimek** [comment]
> > Is my configuration really invalid? No... I guess my branch naming convention is different.
> 
> But you use the branch name to generate the config...That's what the glob do.
> As far as I can tell the error happens when it should and its message is descriptive and accurate.
> I don't really understand what the problem is here?

It doesn't matter. It's just nitpicking in the error wording in specific cases. It's not important to talk more about it.

I have to say that your tone of voice in both PRs isn't welcoming as it should be in the open-source community. Thanks for the experience. I won't participate in any semantic-* libraries.

File: docs/usage/configuration.md [hunk 2]
```diff
@@ -76,6 +76,8 @@ The branches on which releases should happen. By default **semantic-release** wi
 
 **Note**: Once **semantic-release** is configured, any user with the permission to push commits on one of those branches will be able to publish a release. It is recommended to protect those branches, for example with [GitHub protected branches](https://help.github.com/articles/about-protected-branches).
 
+**Note**: The prerelease can't be named with `/` because it doesn't follow the [Semver spec](https://semver.org/). For example `{ "name": "debug/*", "prerelease": true }` won't work. You have to manually change all invalid characters (`/` in this case) to something else (example: `{ "name": "debug/*", "prerelease": '${name.replace(/\//, "-")}' }`)
```
Comment (pvdlg) [inline docs/usage/configuration.md]: As I already mention in https://github.com/semantic-release/semantic-release/pull/1420#issuecomment-576365278 we don't want to mention a workaround specific for the `/` case.
If we have to amend the doc for each possible invalid character a user tries then it's infinite ans we'll never stop changing this doc.

The generated prerelease id has to be valid by the spec and it's up to the user to read the spec. This is why we verify the format in the code and and print an error message linking to the spec.

If users don't read error message and don't click on the spec link there is no much we can do. We can't just list here every imaginable invalid character.

**Comment by gr2m** [comment]
> I have to say that your tone of voice in both PRs isn't welcoming

we are trying to understand your problem and help to the best of our abilities. Neither of us is a native English speaker. We do what we can Prokop, in the little time we have available.

> welcoming as it should be in the open-source community

Your comment sounds very entitled, to be treated the way you think you deserve. I'd appreciate you don't do that, not here or any other Open Source project.

**Comment by pvdlg** [comment]
> It doesn't matter. It's just nitpicking in the error wording in specific cases. It's not important to talk more about it.

I want to improve the doc and I don't understand where it's not clear. That's why I'm trying to figure it out.

> I have to say that your tone of voice in both PRs isn't welcoming as it should be in the open-source community. Thanks for the experience. I won't participate in any semantic-* libraries.

I'm sorry you feel that way and that is not my intention to make you or anyone feel that way.
I challenge your proposed change with a concern for the long term health of the project and documentation. The more we add specific and narrow examples in the doc the more it becomes hard to read for other. In addition, giving examples about narrow cases might lead some users to think it's the recommended approach, while it's actually a case that apply to a very small subset of users.

It's easy to think to "just merge the thing, it's just a line in a doc that might help a few people". But many users will just copy paste that, without realizing it apply only to the narrow cases (using a specific branch naming convention combined with using globs to match pre-release branches) and then later open an issue because that create a problem in their different case.

**Comment by prokopsimek** [comment]
> > It doesn't matter. It's just nitpicking in the error wording in specific cases. It's not important to talk more about it.
> 
> I want to improve the doc and I don't understand where it's not clear. That's why I'm trying to figure it out.
> 

Instead of error `EPRERELEASEBRANCH A pre-release branch configuration is invalid in the branches configuration.` it could be something like `EPRERELEASEBRANCH A pre-release branch name doesn't follow Semver spec.` or `EPRERELEASEBRANCH A pre-release branch has an invalid name.`


> > I have to say that your tone of voice in both PRs isn't welcoming as it should be in the open-source community. Thanks for the experience. I won't participate in any semantic-* libraries.
> 
> I'm sorry you feel that way and that is not my intention to make you or anyone feel that way.
> I challenge your proposed change with a concern for the long term health of the project and documentation. The more we add specific and narrow examples in the doc the more it becomes hard to read for other. In addition, giving examples about narrow cases might lead some users to think it's the recommended approach, while it's actually a case that apply to a very small subset of users.
> 
> It's easy to think to "just merge the thing, it's just a line in a doc that might help a few people". But many users will just copy paste that, without realizing it apply only to the narrow cases (using a specific branch naming convention combined with using globs to match pre-release branches) and then later open an issue because that create a problem in their different case.

Thank for your reply. It would be better just to tell me that `it's actually a case that apply to a very small subset of users`... `please adapt to most`. I just wanted to enhance the documentation to the others who could have the same issue as me. I didn't understand your opinion about the Semver spec the first time. Thank you for your explanation. I believe you do a great job with your libraries according to popularity of your repositories. 🙂

**Comment by pvdlg** [comment]
> Instead of error EPRERELEASEBRANCH A pre-release branch configuration is invalid in the branches configuration. it could be something like EPRERELEASEBRANCH A pre-release branch name doesn't follow Semver spec. or EPRERELEASEBRANCH A pre-release branch has an invalid name.

Ah ok I get it now. The word "configuration" can be misleading when using globs (as the final name is the result of the config+the branch name). We can change that.

**Comment by oocx** [comment]
FYI, I found this thread becase I got the same error and had no idea why. I'd appreciate an error message that makes the reason for the error and potential fixes more obvious.

I also had invalid characters in my branch name and found the solution suggested by @prokopsimek really helpful, thanks!

**Pull request closed without merging**